### PR TITLE
Add table aws_iam_policy_attachment

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  policy_arn,
  is_attached,
  policy_groups
from
  aws_iam_policy_attachment
where
  is_attached;
+----------------------------------------------------------------------------------------------------------------+-------------+----------------------------------------------------------------------+
| policy_arn                                                                                                     | is_attached | policy_groups                                                        |
+----------------------------------------------------------------------------------------------------------------+-------------+----------------------------------------------------------------------+
| arn:aws:iam::632902152528:policy/service-role/AmazonSageMaker-ExecutionPolicy-20210611T124331                  | true        | <null>                                                               |
| arn:aws:iam::632902152528:policy/turbot/admin                                                                  | true        | [{"GroupId":"AGPAZGW7IOFIGY2AUXM6Q","GroupName":"admin"}]            |
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-6fcdb25e-6250-449d-b8da-2399daca9cab | true        | <null>                                                               |
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-fe665362-9c6d-4913-b6f5-cc66c46fd769 | true        | <null>                                                               |
| arn:aws:iam::aws:policy/aws-service-role/AWSOrganizationsServiceTrustPolicy                                    | true        | <null>                                                               |
| arn:aws:iam::aws:policy/AWSElasticBeanstalkMulticontainerDocker                                                | true        | <null>                                                               |
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-b28a9c43-eb11-4b68-91c7-8d26e4029ee2 | true        | <null>                                                               |
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-c7310712-c3cb-4464-b961-4d31f625a816 | true        | <null>                                                               |
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-f613799d-be79-4d31-a2a0-2062b1369cc6 | true        | <null>                                                               |
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-e9a282bc-bcea-4190-8fa1-cff1224d4bb3 | true        | <null>                                                               |
| arn:aws:iam::632902152528:policy/turbot/user                                                                   | true        | [{"GroupId":"AGPAZGW7IOFIMQ4HVH5PI","GroupName":"user"}]             |
| arn:aws:iam::aws:policy/service-role/AmazonDMSVPCManagementRole                                                | true        | <null>                                                               |
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-81bb6720-e3c8-4ec3-bfee-da9aebbd5915 | true        | <null>                                                               |
| arn:aws:iam::aws:policy/service-role/AWSElasticBeanstalkEnhancedHealth                                         | true        | <null>                                                               |
| arn:aws:iam::632902152528:policy/s3acl-allow                                                                   | true        | <null>                                                               |
| arn:aws:iam::aws:policy/AmazonDocDBFullAccess                                                                  | true        | [{"GroupId":"AGPAZGW7IOFIB445OS5TK","GroupName":"capcha6.1"}]        |

> select
  policy_arn,
  is_attached,
  policy_users
from
  aws_iam_policy_attachment
where
  is_attached;
+----------------------------------------------------------------------------------------------------------------+-------------+----------------------------------------------------------------------------
| policy_arn                                                                                                     | is_attached | policy_users                                                               
+----------------------------------------------------------------------------------------------------------------+-------------+----------------------------------------------------------------------------
| arn:aws:iam::632902152528:policy/service-role/AmazonSageMaker-ExecutionPolicy-20210611T124331                  | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/turbot/admin                                                                  | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-6fcdb25e-6250-449d-b8da-2399daca9cab | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-fe665362-9c6d-4913-b6f5-cc66c46fd769 | true        | <null>                                                                     
| arn:aws:iam::aws:policy/aws-service-role/AWSOrganizationsServiceTrustPolicy                                    | true        | <null>                                                                     
| arn:aws:iam::aws:policy/AWSElasticBeanstalkMulticontainerDocker                                                | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-b28a9c43-eb11-4b68-91c7-8d26e4029ee2 | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-c7310712-c3cb-4464-b961-4d31f625a816 | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-f613799d-be79-4d31-a2a0-2062b1369cc6 | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-e9a282bc-bcea-4190-8fa1-cff1224d4bb3 | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/turbot/user                                                                   | true        | <null>                                                                     
| arn:aws:iam::aws:policy/service-role/AmazonDMSVPCManagementRole                                                | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-81bb6720-e3c8-4ec3-bfee-da9aebbd5915 | true        | <null>                                                                     
| arn:aws:iam::aws:policy/service-role/AWSElasticBeanstalkEnhancedHealth                                         | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/s3acl-allow                                                                   | true        | <null>                                                                     
| arn:aws:iam::aws:policy/AmazonDocDBFullAccess                                                                  | true        | <null>                                                                     
| arn:aws:iam::aws:policy/service-role/AWSBackupServiceRolePolicyForBackup                                       | true        | <null>                                                                     
| arn:aws:iam::aws:policy/aws-service-role/AWSSSOServiceRolePolicy                                               | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-68a7b14e-801a-4f84-ba15-7a41b7ad20fc | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/service-role/KinesisFirehoseServicePolicy-aws-waf-logs-test001-us-east-1      | true        | <null>                                                                     
| arn:aws:iam::aws:policy/aws-service-role/WAFRegionalLoggingServiceRolePolicy                                   | true        | <null>                                                                     
| arn:aws:iam::aws:policy/aws-service-role/AWSElasticBeanstalkServiceRolePolicy                                  | true        | <null>                                                                     
| arn:aws:iam::aws:policy/service-role/CloudWatchEventsBuiltInTargetExecutionAccess                              | true        | <null>                                                                     
| arn:aws:iam::632902152528:policy/service-role/s3crr_for_capcha7.2_e891c0                                       | true        | <null>                                                                     
| arn:aws:iam::aws:policy/AdministratorAccess                                                                    | true        | [{"UserId":"AIDAZGW7IOFIAW2XPYQT7","UserName":"partha-console"},{"UserId":"
| arn:aws:iam::632902152528:policy/service-role/AWSLambdaBasicExecutionRole-4d51f6f8-81cb-43fc-b593-d902079e5bad | true        | <null>                                                                     
| arn:aws:iam::aws:policy/AmazonMacieFullAccess                                                                  | true        | [{"UserId":"AIDAZGW7IOFIIRLMKNIZR","UserName":"sourav"}]                   

> select
  name as policy_name, 
  policy_arn, 
  jsonb_pretty(policy_users) as policy_users
from
  aws_iam_policy p
  left join aws_iam_policy_attachment a on p.arn = a.policy_arn 
where
  name = 'AdministratorAccess' and a.is_attached;
+---------------------+---------------------------------------------+--------------------------------------------+
| policy_name         | policy_arn                                  | policy_users                               |
+---------------------+---------------------------------------------+--------------------------------------------+
| AdministratorAccess | arn:aws:iam::aws:policy/AdministratorAccess | [                                          |
|                     |                                             |     {                                      |
|                     |                                             |         "UserId": "AIDAZGW7IOFIAW2XPYQT7", |
|                     |                                             |         "UserName": "partha-console"       |
|                     |                                             |     },                                     |
|                     |                                             |     {                                      |
|                     |                                             |         "UserId": "AIDAZGW7IOFICJWU64JEF", |
|                     |                                             |         "UserName": "steampipe-cloud"      |
|                     |                                             |     },                                     |
|                     |                                             |     {                                      |
|                     |                                             |         "UserId": "AIDAZGW7IOFIJCQKJQ3N5", |
|                     |                                             |         "UserName": "raj"                  |
|                     |                                             |     },                                     |
|                     |                                             |     {                                      |
|                     |                                             |         "UserId": "AIDAZGW7IOFINSYQZCGS7", |
|                     |                                             |         "UserName": "user-steampipe"       |
|                     |                                             |     }                                      |
|                     |                                             | ]                                          |
+---------------------+---------------------------------------------+--------------------------------------------+

```
</details>

### CODE REVIEW

File: docs/tables/aws_iam_policy_attachment.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Table: aws_iam_policy_attachment
+
+This table provide information about the attached user(s), role(s), and group(s) to a Managed IAM Policy
```
Comment (misraved) [inline docs/tables/aws_iam_policy_attachment.md:3]:
```suggestion
This table provides information about the attached user(s), role(s), and group(s) to a Managed IAM Policy.
```

File: aws/table_aws_iam_policy_attachment.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+			ParentHydrate: listIamPolicies,
+			Hydrate:       listIamPolicyAttachments,
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "is_attached", Require: plugin.Optional, Operators: []string{"<>", "="}},
+			},
+		},
+		Columns: awsColumns([]*plugin.Column{
+			{
+				Name:        "policy_arn",
+				Description: "The Amazon Resource Name (ARN) specifying the iam policy.",
```
Comment (misraved) [inline aws/table_aws_iam_policy_attachment.go:30]:
```suggestion
				Description: "The Amazon Resource Name (ARN) specifying the IAM policy.",
```

File: aws/table_aws_iam_policy_attachment.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	// List call
+	err = svc.ListEntitiesForPolicyPages(params, func(page *iam.ListEntitiesForPolicyOutput, lastPage bool) bool {
+		d.StreamListItem(ctx, PolicyAttachment{*policy.Arn, policy.AttachmentCount, page.PolicyGroups, page.PolicyRoles, page.PolicyUsers})
+
+		// Check if context has been cancelled or if the limit has been hit (if specified)
+		// if there is a limit, it will return the number of rows required to reach this limit
+		if d.QueryStatus.RowsRemaining(ctx) == 0 {
+			return false
+		}
+
```
Comment (misraved) [inline aws/table_aws_iam_policy_attachment.go:103]:
```suggestion
```

File: aws/table_aws_iam_policy_attachment.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+		return !lastPage
+	},
```
Comment (misraved) [inline aws/table_aws_iam_policy_attachment.go:105]: Is the indentation correct?

**Review by ParthaI — APPROVED** [review]

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**